todo_app/tests/test_mocking.py

```diff
@@ -0,0 +1,34 @@
+import os
+import pytest
+import mongomock
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+from pymongo import MongoClient
+
+@pytest.fixture
+def client(monkeypatch):
+    file_path = find_dotenv(".env.test")
+    load_dotenv(file_path, override=True)
+
+    mock_client = mongomock.MongoClient()
+    monkeypatch.setattr(MongoClient, "__init__", lambda self, *args, **kwargs: None)
+    monkeypatch.setattr(MongoClient, "get_database", lambda self, *args, **kwargs: mock_client['todo_app_db'])
+
+    test_app = app.create_app()
+    with test_app.test_client() as client:
+        yield client
+
+
+def test_index_page(client):
+    db = MongoClient().get_database("todo_app_db")  # The mocked database
```

Comment:
Note here that this doesn't make it clear how it matches the DB the app actually tries to use - I can believe your mocking works, but I think it would be nicer to be consistent!
```suggestion
    MONGO_URI = os.getenv('MONGO_CONNECTION_STRING')
    DATABASE_NAME = os.getenv('DATABASE_NAME')
    db = pymongo.MongoClient(MONGO_URI).get_database(DATABASE_NAME)
```